pkg/webhook/external/artifactory.go

```diff
@@ -148,41 +149,51 @@ func (a *artifactoryWebhookReceiver) getHandler(requestBody []byte) http.Handler
 			return
 		}
 
-		originURL, err := url.Parse(payload.Origin)
-		if err != nil {
+		if payload.Data.ImageType != artifactoryDockerDomain && payload.Data.ImageType != artifactoryChartImageType {
 			xhttp.WriteErrorJSON(
 				w,
-				xhttp.Error(errors.New("invalid request body"), http.StatusBadRequest),
+				xhttp.Error(
+					fmt.Errorf("unsupported image type %q", payload.Data.ImageType),
+					http.StatusNotImplemented,
+				),
 			)
 			return
 		}
 
-		pathSections := strings.Split(payload.Data.Path, "/")
-		repoURL := strings.Join(
-			append(
-				[]string{originURL.Host, payload.Data.RepoKey},
-				pathSections[:len(pathSections)-2]...,
-			),
-			"/",
-		)
-
 		var repoURLs []string
-		switch payload.Data.ImageType {
-		case artifactoryDockerDomain:
-			repoURLs = append(repoURLs, urls.NormalizeImage(repoURL))
-		case artifactoryChartImageType:
-			repoURLs = append(repoURLs, urls.NormalizeChart(repoURL))
-		default:
-			xhttp.WriteErrorJSON(
-				w,
-				xhttp.Error(
-					fmt.Errorf("unsupported image type %q", payload.Data.ImageType),
-					http.StatusNotImplemented,
+		if repoURLsHeader := r.Header.Get(artifactoryRepoURLHeader); repoURLsHeader != "" {
+			repoURLs = strings.Split(repoURLsHeader, ",")
```

Comment:
Should these URLs still go through validation (e.g. `url.Parse`)? Additionally, should we limit the number of allowed repository URLs?